decoder/decoder.go

```diff
@@ -42,21 +65,33 @@ func readRequestJSONData(maxSize int64) Reader {
 			return nil, errors.New("no content")
 		}
 
+		cLen := req.ContentLength
+		knownCLen := cLen > -1
 		switch req.Header.Get("Content-Encoding") {
 		case "deflate":
+			if knownCLen {
+				deflateLengthAcc.Add(cLen)
+			}
 			var err error
 			reader, err = zlib.NewReader(reader)
 			if err != nil {
 				return nil, err
 			}
 
 		case "gzip":
+			if knownCLen {
+				gzipLengthAcc.Add(cLen)
+			}
 			var err error
 			reader, err = gzip.NewReader(reader)
 			if err != nil {
 				return nil, err
 			}
 		}
+		if knownCLen {
+			uncompressedLengthAcc.Add(cLen)
```

Comment:
What do you think about moving this into the `switch` `default` section to only cover non-compressed requests and not have an overall counter?

Reply:
there isn't an overall counter, all the switch cases return from the function. but i can put this a a default.

Reply:
actually you are right. i don't know what i am saying.